### ISSUE TITLE
Hi. Is there a way to get the list of subscribed plans for the user.

### ISSUE THREAD

**Issue by ankurToko** [dooboolab/react-native-iap#17]
Hi. Is there a way to get the list of subscribed plans for the user so that we do not display the subscription UI if the user is already on a paid plan.

**Comment by dooboolab** [comment]
@ankurToko 
sure you can in Android but ios, I think it is not fully supported at the moment. @JJMoon will answer your question soon.

**Comment by ankurToko** [comment]
@dooboolab @JJMoon  Yeah, I had the ios usecase. Any work around or solution?

**Comment by JJMoon** [comment]
Please refer the Apple document : https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/StoreKitGuide/Chapters/Subscriptions.html

I will implement regarding this in the near future.

**Comment by dooboolab** [comment]
@ankurToko @JJMoon 
`getPurchaseItemsAndroid(inapp : string)`
We are missing above method currently for IOS.
Also looking for the contributor!

**Comment by ankurToko** [comment]
Hi @JJMoon @dooboolab 

Is there some breaking change in the RNIapIos.m file? The code that was working earlier now shows exception here

`NSDictionary *dic = @{ @"productId" : theProd.productIdentifier,
                         @"price" : theProd.price,
                         @"currency" : theProd.priceLocale.currencyCode,
                         @"title" : theProd.localizedTitle,
                         @"description" : theProd.localizedDescription,
                         @"localizedPrice" : localizedPrice
                         };
`

This is the exception

`*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[3]'
*** First throw call stack:
(0x1924befe0 0x190f20538 0x1923a59b4 0x1923a5824 0x10070ca98 0x10070b578 0x19f58d320 0x10106da50 0x10106da10 0x101072b78 0x19246d0c8 0x19246ace4 0x19239ada4 0x193e05074 0x19864ec9c 0x100021054 0x1913a959c)
libc++abi.dylib: terminating with uncaught exception of type NSException`

>

**Comment by dooboolab** [comment]
@ankurToko 
It would be better if you paste your js codes here too so we can tell what is going wrong.

**Comment by ankurToko** [comment]
@dooboolab  There you go. Just the simple buy stuff.

`RNIap.getItems(itemSkus).then(items=> {
      console.log("get items called: ", items);
      RNIap.buyItem('xyz').then(receipt=> {
        console.log("bought items: ", receipt);
      }).catch(errorCode=>{
        console.log("ErrorCode: ", errorCode)
      })}
    ).catch(error=>{
console.log("Error: ", error)
    })`

**Comment by dooboolab** [comment]
@ankurToko 
Is item called? I mean can you see the logs? If you do, in which line does the crash occurs?
If you got crashed in `RNIap.getItems(itemSkus)` then I suspect your item has not been set properly in [itunesconnect]('itunesconnect.apple.com'). Could you check that?

**Comment by ankurToko** [comment]
@dooboolab 
I get the correct valid product count and the product ids get listed correctly. Also earlier it was working fine, but then recently i tried upgrading the library, but that didnt work out. After that when I tried rolling back to the older version that I was using, it showed this error to me. Earlier I was using 0.1.10

**Comment by dooboolab** [comment]
@ankurToko 
If you see the Readme, you can see what has been changed critically.
How is your import statement done?
Just said, we need more information to guide you.
Please review my previous comment too.

**Comment by ankurToko** [comment]
@dooboolab 

**Import statement** is `import * as RNIap from 'react-native-iap';`

**The code stops here**
`NSDictionary *dic = @{ @"productId" : theProd.productIdentifier,` in RNIapIos.m

**My itemSkus is**
```
const itemSkus = {
  ios: [
    'xyz',
    'abc',
    'pqr',
  ],
};

```

My in-app products in the iTunes are correctly configured as i get them in the valid Product count in the logs and also it used to work until i tried upgrading.

Currently I am on 0.2.0

**Comment by dooboolab** [comment]
@ankurToko 
Looks you've updated to `0.2.*` correctly.
Can you tell where it exactly is crashing? Is it when you buyItem or when you getItems?

**Comment by JJMoon** [comment]
Are 'xyz', 'abc' product IDs? You should not use Identification Infomation. And usually the ID use same notation with bundle id, like prod.second.mysite.com etc

**Comment by ankurToko** [comment]
@JJMoon I have just masked those values. And as mentioned earlier, it was working earlier and I was able to buy the product in my sandbox environement.

@dooboolab it is crashing in the getItems. The logs start coming 

`2018-01-30 12:51:48.196 [info][tid:com.facebook.react.RNIapIosQueue][RNIapIos.m:46] 



 Obj c >> InAppPurchase  :: fetchProducts 



 .
2018-01-30 12:51:48.196370+0530 APP NAME[772:191753] 



 Obj c >> InAppPurchase  :: fetchProducts 



 .
2018-01-30 12:51:48.198672+0530 APP NAME[772:191753]    Obj c >> InAppPurchase  :: fetchProducts  >>  arrProd parsed as follows  ... (
    "abc",
    "pqr",
    "xyz"
)
2018-01-30 12:51:49.267226+0530 APP NAME [772:191710] 


 Obj c >> InAppPurchase   ###  didReceiveResponse :: Valid Product Count ::  3
2018-01-30 12:51:49.270598+0530 APP NAME[772:191710] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[3]'
*** First throw call stack:
(0x1924befe0 0x190f20538 0x1923a59b4 0x1923a5824 0x1007b8a98 0x1007b7578 0x19f58d320 0x101209a50 0x101209a10 0x10120eb78 0x19246d0c8 0x19246ace4 0x19239ada4 0x193e05074 0x19864ec9c 0x1000cd054 0x1913a959c)
libc++abi.dylib: terminating with uncaught exception of type NSException`

**Comment by dooboolab** [comment]
@ankurToko 
It is quite inefficient to solve your issue here because there hasn't been chages from `0.1*` to `0.2.*` for getting items in ios. So we need to try some test and trials to find out your problem. Could please getItems one by one for your ios products?
```
const itemSkus = {
  ios: [
    'xyz',
//    'abc',
//    'pqr',
  ],
};
```
one by one. It might not be the issue but need to do some test yourself because we are not in your environment..

**Comment by ankurToko** [comment]
@dooboolab 
Alright. Will be on it.

**Comment by dooboolab** [comment]
@ankurToko 
Please tell us again if you are still in trouble or any experience!

**Comment by ankurToko** [comment]
@dooboolab 
I still haven't been able to resolve it. Trying upgrading. The single item thing didn't work out.

**Comment by dooboolab** [comment]
@ankurToko 
Could you try example code in [here](https://github.com/dooboolab/react-native-iap/tree/master/RNIapExample)?

I will invite you to this itunesconnect test project if you share your apple id. If you feel private with this one you can send me an email([dooboolab@gmail.com](mailto:dooboolab@gmail.com)).

Maybe you can compare with your configuration in itunesconnect and sample codes.

**Comment by ankurToko** [comment]
@dooboolab @JJMoon 
I rectified the issue that I had. It was because of 0.1.15 As in the package.json there was a '^' it was taking the latest version of 0.1.15, but when i removed the '^' then it remained on 0.1.10 and worked just like before.

I suppose this has something to do with any of your latest updates.

**Comment by dooboolab** [comment]
@ankurToko 
Glad that it is solved. I've told you previously that there is critical bug in `0.1.15` that @JJMoon  born. Good that you found out. Also, I've created another issue for previous question [here](https://github.com/dooboolab/react-native-iap/issues/18). Because there are mixed personality in this issue thread.

I've just seperated bugs with getting the list of subscribed plans for the user.

Thank you. I will close this.
